packages/runtime/test/integration/group-inner-transaction.ts

```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
 		// spend 5000 micro Algo
 		assert.equal(contractBalance, contractAcc.balance() + 5000n);
 	});
+
+	describe("Number inner transaction limitation", function () {
+		let secondAppInfo: AppInfo;
+		this.beforeEach(() => {
+			secondAppInfo = runtime.deployApp(
+				alice.account,
+				{
+					appName: "secondApp",
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: "limit-number-txn.py",
+					clearProgramFilename: "clear.teal",
+					globalBytes: 1,
+					globalInts: 1,
+					localBytes: 1,
+					localInts: 1,
+				},
+				{}
+			);
+
+			const txnParams: types.AlgoTransferParam = {
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: secondAppInfo.applicationAccount,
+				amountMicroAlgos: 1e6,
+				payFlags: {
+					totalFee: 1000,
+				},
+			};
+
+			runtime.executeTx([txnParams]);
+			syncAccounts();
+		});
+
+		it("Should fail when issue more than 256 inner txn", () => {
+			const firstTxn: types.ExecParams = {
+				type: types.TransactionType.CallApp,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				appID: secondAppInfo.appID,
+				appArgs: ["str:exec"],
+				payFlags: {
+					totalFee: 1000,
+				},
+			};
+			const secondTxn: types.ExecParams = {
+				type: types.TransactionType.CallApp,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				appID: secondAppInfo.appID,
+				appArgs: ["str:exec"],
+				payFlags: {
+					totalFee: 1000,
+					note: "second",
+				},
+			};
```

Comment:
We should use a function for this. In general if you're repeating any code more than 3 times, you should avoid it. You can create a function

Reply:
Yeah, good idea!